app/scripts/background.js

```diff
@@ -482,6 +504,26 @@ export function setupController(initState, initLangCode, overrides) {
    * @param {Port} remotePort - The port provided by a new context.
    */
   connectRemote = async (remotePort) => {
+    ///: BEGIN:ONLY_INCLUDE_IN(desktop)
+    if (
+      DesktopManager.isDesktopEnabled() &&
+      OVERRIDE_ORIGIN.DESKTOP !== overrides?.getOrigin?.()
+    ) {
+      DesktopManager.createStream(remotePort, CONNECTION_TYPE_INTERNAL).then(
+        () => {
+          // When in Desktop Mode the responsibility to send CONNECTION_READY is on the desktop app side
+          if (isManifestV3) {
+            // Message below if captured by UI code in app/scripts/ui.js which will trigger UI initialisation
+            // This ensures that UI is initialised only after background is ready
+            // It fixes the issue of blank screen coming when extension is loaded, the issue is very frequent in MV3
```

Comment:
"It fixes the issue of blank screen coming when extension is loaded, the issue is very frequent in MV3"

And so I assume that the issue does not happen in MV2, is that correct?

Reply:
Yes this post message is not needed in MV2 and thus may not be needed in desktop also.